packages/admin/admin-vite-plugin/src/routes/generate-menu-items.ts

```diff
@@ -107,70 +113,58 @@ function generateImport(file: string, index: number): string {
 }
 
 function generateMenuItem(
-  config: { label: boolean; icon: boolean },
+  config: RouteConfig,
   file: string,
   index: number
 ): MenuItem {
   const configName = generateRouteConfigName(index)
-  const routePath = getRoute(file)
-
   return {
     label: `${configName}.label`,
     icon: config.icon ? `${configName}.icon` : undefined,
-    path: routePath,
+    path: getRoute(file),
+    nested: config.nested
   }
 }
 
-async function getRouteConfig(
-  file: string
-): Promise<{ label: boolean; icon: boolean } | null> {
+async function getRouteConfig(file: string): Promise<RouteConfig | null> {
   const code = await fs.readFile(file, "utf-8")
-
   let ast: ParseResult<File> | null = null
 
   try {
     ast = parse(code, getParserOptions(file))
   } catch (e) {
-    logger.error(`An error occurred while parsing the file.`, {
-      file,
-      error: e,
-    })
+    logger.error(`An error occurred while parsing the file.`, { file, error: e })
     return null
   }
 
-  let config: { label: boolean; icon: boolean } | null = null
+  let config: RouteConfig | null = null
 
   try {
     traverse(ast, {
       ExportNamedDeclaration(path) {
         const properties = getConfigObjectProperties(path)
+        if (!properties) return
 
-        if (!properties) {
-          return
-        }
-
-        const hasLabel = properties.some(
-          (prop) =>
-            isObjectProperty(prop) && isIdentifier(prop.key, { name: "label" })
+        const hasProperty = (name: string) => properties.some(
+          (prop) => isObjectProperty(prop) && isIdentifier(prop.key, { name })
         )
 
-        if (!hasLabel) {
-          return
-        }
+        const hasLabel = hasProperty("label")
+        if (!hasLabel) return
 
-        const hasIcon = properties.some(
-          (prop) =>
-            isObjectProperty(prop) && isIdentifier(prop.key, { name: "icon" })
+        const nested = properties.find(
```

Comment:
todo: We should validate that the value is one of the allowed ones.

Reply:
Added validation